### ISSUE TITLE
Error on Ubuntu OS

### ISSUE THREAD

**Issue by omkarnathsingh** [TGMTProto/MTProtoProxy#15]
`Error receiving: System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values.
Parameter name: offset
   at System.Net.Sockets.SocketAsyncEventArgs.SetBuffer(Byte[] buffer, Int32 offset, Int32 count)
   at MTProtoProxy.SocketUtils.ReceiveAsync(Socket socket, Byte[]& buffer, Int32& offset, Int32& count)
   at MTProtoProxy.MTPSocket.ReceiveAsync(Byte[] buffer)
   at MTProtoProxy.MTPSocket.ReceiveAsync()
   at MTProtoProxy.MTPSocket.StartReceiverAsync()
`

I am getting above errors in between, btw thanks, latest dll working on Linux Ubuntu OS. Thanks.

**Comment by TGMTProrto** [comment]
Use the new version(1.0.5-alpha)

**Comment by p1ratrulezzz** [comment]
Hi. On ubuntu I have

```
SHA256Helper.cs(7,45): error CS8107: Feature 'readonly references' is not available in C# 7.0. Please use language version 7.2 or greater. 
```

A lot of such errors

Sorry I'm not a C# expert and don't know all the nuances. Maybe also would be great to add default Program.cs with Main() function to git?
